Add documentation for Finalize/Abort failed (#316)
openshift-kni/lifecycle-agent: docs/image-based-upgrade.md

Once completed, the system is ready for the next upgrade.

#### Finalize or Abort failure

If finalize or abort fails, IBU transitions into `FinalizeFailed` or
`AbortFailed` states respectively.

```yaml
  status:
    conditions:
    - message: failed to cleanup backups. Perform cleanup manually then
add lca.openshift.io/manualCleanupDone
        annotation to ibu CR to transition back to Idle
      observedGeneration: 5
      reason: AbortFailed
      status: "False"
      type: Idle
```

The condition message indicates which parts of cleanup has failed.
User should perform the cleanup manually in these states.
This mostly involves making sure that old stateroot is cleaned up and
all backup resources are removed.
Backup resources are `backups.velero.io`, `restores.velero.io` and `deletebackuprequests.velero.io`.

> After manual cleanup user should add 'lca.openshift.io/manualCleanupDone' annotation

Reviewer: could you link the steps to manual cleanup?

Author: I updated the PR. let me know if it makes sense.

Once completed, the system is ready for the next upgrade.

#### Finalize or Abort failure

If finalize or abort fails, IBU transitions into `FinalizeFailed` or
`AbortFailed` states respectively.

```yaml
  status:
    conditions:

>     - message: failed to cleanup backups. Perform cleanup manually then
> add lca.openshift.io/manualCleanupDone
>         annotation to ibu CR to transition back to Idle

Reviewer 2: ```suggestion
    - message: failed to delete all the backup CRs. Perform cleanup manually then add 'lca.openshift.io/manualCleanupDone' annotation to ibu CR to transition back to Idle
```

      observedGeneration: 5
      reason: AbortFailed
      status: "False"
      type: Idle
```

> The condition message indicates which parts of cleanup has failed.

Reviewer 2: ```suggestion
The condition message indicates which parts of the cleanup have failed.
```

User should perform the [cleanup manually](troubleshooting.md#manual-cleanup) in these states.
After manual cleanup user should add `lca.openshift.io/manualCleanupDone` annotation
to IBU CR.

```yaml
  kind: ImageBasedUpgrade
  metadata:
    annotations:
      lca.openshit.io/manualCleanupDone: ""

```

> Upon observing this annotation IBU will remove the annotation and runs

Reviewer 2: ```suggestion
Upon observing this annotation, IBU will remove the annotation and run
```
